docs/build_stage_compile_settings.md

```diff
@@ -0,0 +1,33 @@
+# Compile Settings for Build Stages
```

Comment:
I am generally dubious that we will keep this document up to date. It seems better to have a single source of truth (the Jenkinsfile that runs). Can you push this to the Jenkinsfile, define an array or something?

Reply:
I think defining an array semi defeats the purpose of this. Which was to make it so change to the build would only require a single change. I included a last updated line in the docs because I anticipated that issue as well. I'm not entirely sure the docs are necessary.

Another potential solution would be to build documentation rules into the build. AKA if the Jenkins-utils file changes and this one does not then fail the build. (That might be a little strict but there could be good rules to enforce documentation)

Reply:
Another problem of defining an array is then there is a lack of transparency into the build from the Jenkinsfile. Currently, by passing in just the args that pertain to the stage we can see what is most important/unique about the build for that stage.

Reply:
I don't follow, sorry. For example, if you define a base config `C` with all the defaults and then each entry of the array is just `C.SetParam1(true).SetParam2(false)`, etc., that seems to be the best of both worlds.

As it stands in this PR, a change to the build system requires changing all the files mentioned above including the documentation here.

Reply:
Can you checkout the changes I pushed to `Jenkinsfile-utils`. I'm kinda doing that but not in an array. I have a default config and then rules that apply the changes. I think one question is whether the a developer needs to be aware of the distinct aspects of the build when looking at the Jenkinsfile/Jenkinsfile-nightly. If they do then the current solution seems good (if I delete the docs) because it highlights the key attributes as arguments to the function called in the Jenkinsfile/Jenkinsfile-nightly (i.e. `utils.noisePageBuild(buildType:utils.RELEASE_BUILD, isBuildTests:false)`). However, if when changing the Jenkinsfile the developer does not need to know about the build config then we could make that method `utils.noisePageBuild('End-to-End Performance')` and then do a map of stage->config in this utils. Thoughts?